### PR TITLE
Add goal status graph widget on backend

### PR DESCRIPTION
**Pull request by jasalisbury**
## Description of change

Needed to add two missing scopes to goals, region and recipient id. Added some test utilities to create/destroy goals to aid in testing. Check pointing at the backend so someone else can pickup the ticket and do the frontend as I won't have time to work on the frontend for a few days.

## How to test

Make sure tests make sense and pass.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-456

### Every PR

- [ ] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/widgets/goalStatusGraph.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+} from '../models';
+
+export default async function goalStatusGraph(scopes) {
+  const goals = await Goal.findAll({
+    where: {
+      [Op.and]: [
+        scopes.goal,
+        {
+          status: {
+            [Op.not]: null,
```
Comment (thewatermethod) [inline src/widgets/goalStatusGraph.js:13]: Elsewhere, we've been using "needs status" if the goal was created in the hub and therefore doesn't have a status. Do you think we should use it here?

File: src/widgets/index.js [hunk 2]
```diff
@@ -19,4 +20,5 @@ export default {
   topicFrequencyGraph,
   targetPopulationTable,
   frequencyGraph,
+  goalStatusGraph,
```
Comment (thewatermethod) [inline src/widgets/index.js:23]: Probably not a huge deal, but querying this API returns 238 goals with a status of "" on my DB. Possibly something weird from imported data?

Screen Shot 2022-02-09 at 7 59 19 AM

File: src/widgets/goalStatusGraph.js (hunk 1 above)
Comment (jasalisbury) [reply src/widgets/goalStatusGraph.js:13]: "needs status" and "draft" goals are ignored in the widget. I'm updating the backend to explicitly ignore draft goals

File: src/widgets/index.js (hunk 2 above)
Comment (jasalisbury) [reply src/widgets/index.js:23]: Yeah those goals come in from smartsheet as empty strings. I'd imagine we'd want to update those goals to have a status that matches "needs status" (either null or explicitly "needs status")

**Review by thewatermethod — APPROVED** [review]

File: src/widgets/goalStatusGraph.js [hunk 3]
```diff
@@ -3,14 +3,29 @@ import {
...
 export default async function goalStatusGraph(scopes) {
-  const goals = await Goal.findAll({
+  const goalsFromDb = await Goal.findAll({
     where: {
       [Op.and]: [
         scopes.goal,
         {
           status: {
-            [Op.not]: null,
+            [Op.in]: STATUSES_TO_INCLUDE,
```
Comment (thewatermethod) [inline src/widgets/goalStatusGraph.js:28]: ✨

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**